### DISCUSSION TITLE
A comparison with Vim keybindings on basic operations

### DISCUSSION THREAD

**Discussion by DogLooksGood** [meow-edit/meow#101] [General]
https://gist.github.com/DogLooksGood/559e796b8fd913771e083d969b620e70

A draft, WIP

**Comment by Vincent-P** [comment]
Hello,
I am a long time vim/evil user and I decided to try meow this morning.
After finishing meow-tutor and tried to code for a bit I still haven't found some vim features:
- `{` and `}`: I think it should be `[p` and `]p`? However `[p` doesn't seem to work on my setup I can only go forward with `]p`. Is there a way to extend the selection? For example if you have 2 paragraphs and you press `]p` twice, it selects the first paragraph then the second one. Let's say I want to select **both** of them how do I do that?
- `C-u`, `C-d` move half a screen up or down, am I supposed to use the emacs keybindings for that? `C-v` and `M-v`?

**Comment by DogLooksGood** [comment]
For the first question,
Currently, paragraph support in meow is just thing based selection. The only way you can select two paragrah at the moment, is to create a selection, then use emacs paragraph movement(forward is bound to `M-}` and `C-<down>` by default). Whatever selection is ok for this case, since Emacs's built-in command won't cancel the selection. It's easy to add a standalone paragraph command. but I'm still considering whether it is the best way or not. The same problem for sentence support. An non-equivalent solution is select line, then expand by numbers, the case is different from Vim since you don't have to count the numbers before you select, just follow the hint.

I guess our solution will be one of following:
- add nothing, instead use word/line/find
- add paragraph, sentence movement command
- allow thing-based command expanable via numbers(but "select two" feels much better than "select 1 + expand 1")

For the second question,
Yes, use emacs built-in solutions. Meow doesn't care about paging. There is one built-in solution(probably there's more)
```
(require 'view) ;; built-in
```
Then use `View-scroll-half-page-forward/backward`.

**Reply by Vincent-P** [reply]
Thanks for your answer!
The third solution you mentioned is the first thing I tried, pressing 1 to extend the selection made sense because that's how other selections work.
